packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariables.ts

```diff
@@ -0,0 +1,148 @@
+import path from 'path';
```

Comment:
I think for all of the functions in this file we only need "set" and "delete" operations (I would use "set" instead of "add" because if you try to add an env var with the same name it should overwrite the previous one, not add a new one. So basically I think the exported interface of this module should be something like:
```ts
export const setEnvVar = (name: string, value: string): void;

export const deleteEnvVar = (name: string);
```

Under the hood their should be non-exported functions for setting / removing the relevant config from team-provider-info, function-parameters and the cfn template. What you have is close to this, it just seems a bit over complicated.

Reply:
I agree that we may not need update function to export from this file. However, we should have `getStoredEnvironmentVariables` function which is used in 3 places. To not hard code `deploymentBucketName` and `s3Key`, we need a matching alrogithm([refs](https://github.com/aws-amplify/amplify-cli/blob/8166855fd74b66bb1bc394719d7ad8796c49a642/packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariables.ts#L83-L99)) and it's actually a bit complicated.

JFYI - In addition, I also consider if we can have only `syncEnvVar` function to custom environment variables, but turned out it's complicated to not touch `deploymentBucketName` and `s3Key` for the same reson because it requires diff detection alrogthm.

Reply:
I just added validations to take care overwriting. Please find the attached screenshots of `Cannot add new environment variable with duplicated KEY` and `Cannot edit existed environment variable when it is not selected` in the description to see their experiences.